packages/react-utils/src/compose/types.ts

```diff
@@ -35,7 +34,7 @@ export type ShorthandProps<TProps extends ComponentProps = {}> =
         children?: TProps['children'] | ShorthandRenderFunction<TProps>;
       });
 
-export type ObjectShorthandProps<TProps extends GenericDictionary> = TProps & {
+export type ObjectShorthandProps<TProps extends ComponentProps> = TProps & {
```

Comment:
```suggestion
export type ObjectShorthandProps<TProps extends ComponentProps = {}> = TProps & {
```

To be aligned to `ShorthandProps` type for now